examples/cxx/publish_subscribe_cross_language/README.md

```diff
@@ -0,0 +1,136 @@
+# Cross-language Publish-Subscribe
+
+Before proceeding, all dependencies need to be installed. You can find
+instructions in the [C++ Examples Readme](../README.md).
+
+## Running The Example
+
+> [!CAUTION]
+> Every payload you transmit with iceoryx2 must be compatible with shared
+> memory. Specifically, it must:
+>
+> * be self contained, no heap, no pointers to external sources
+> * have a uniform memory representation, ensuring that shared structs have the
+>     same data layout
+> * not use pointers to manage their internal structure
+>
+> Data types like `std::string` or `std::vector` will cause undefined behavior
+> and may result in segmentation faults. We provide alternative data types
+> that are compatible with shared memory. See the
+> [complex data type example](../complex_data_types) for guidance on how to
+> use them.
+
+This example illustrates a robust publisher-subscriber communication pattern
```

Comment:
Please, a bit more language agnostic please so that we do not need to modify the readme when we add a new language binding that can be used for communication as well in this scenario.